### DISCUSSION TITLE
Project Migration Feature Preview

### DISCUSSION THREAD

**Discussion by j0siepy** [community/community#18752] [Projects and Issues] [answered]
You can now migrate your old project (classic) boards to the all new projects (beta) experience! Enable "Project migration" in feature preview to try out tables, multiple views, new automation options, and powerful field types using your existing project (classic) boards. 

How it works:
- We'll create a copy of your existing project (classic) board in Projects (beta)
- You'll be prompted to close your our old project (classic) as updates made in the new project (Beta) will not be reflected in the closed project (classic)
- You can reopen the classic project in a pinch if necessary! 

Leave your feedback below!

**Comment by Katsute** [comment]
Two criticisms that I have about the current state of project tables:
1. **No repository level projects** - https://github.com/github-community/community/discussions/6229
   All of my projects are exclusively for the repositories that they reside in, it makes no sense to make them all organization-level or user-level projects. The classic projects are available for repository-level, so why aren't project tables? 
2. **No markdown cards** - https://github.com/github-community/community/discussions/4422
   Draft issues aren't a viable replacement for markdown cards, rather than seeing all the cards contents once, we are forced to click each individual issue to see it's contents. One line titles convey significantly less information than a markdown card.

Both of these features were on the original roadmap for projects beta, why were they removed when it's a standard feature in projects classic?

**Reply by nickserv** [reply]
You can still link projects to your repository, but since they're account-level you can also include cards from other repositories if you want to.

**Reply by wwkimball** [reply]
There is just no scenario where I would want "cards from other repositories" in my repository-level Project.  The scope of my projects is bound inextricably to the repository I created the project for.

**Comment by haykam821** [comment]
The migration popup indicates that "Automation, activity, archive, and triage won’t be migrated". Are these features coming to a future version of project migration?

In addition, will events on issues link to the migrated projects (beta) board or to the projects (classic) board?

**Reply by j0siepy** [reply]
Archive will now be migrated - but automation, activity and triage won't be.

**Comment by morgzz** [comment, marked as answer]
Spotted a spelling mistake when enabling feature preview:

> You'll be prompted to close **_your our_** old project as the newly migrated project and old project will not be kept in sync

**Reply by j0siepy** [reply]
Fixed, Thank you!

**Reply by ghost** [reply]
امم

**Reply by ziljah** [reply]
What should I do after migrating the project? Are you done ?

**Comment by nickserv** [comment]
Some of my more complex projects still fail to import, but the error message doesn't explain why.

![Something went wrong: Your project could not be migrated.](https://user-images.githubusercontent.com/927220/180246893-911625c2-9a5e-4c8d-8354-e4b955c484d5.png)

**Reply by j0siepy** [reply]
Are you still running into this issue?

**Comment by nervewax** [comment]
- It would be handy if there was a link from the old (migrated) board to the new one in the warning notice displayed at the top of the old board.
- When notes are migrated to drafts on the new project, the first line of the note is used as the title. If the first line could also be output in the content of the draft too, that would make more sense, to match the experience of the old board being able to copy the entire contents of the note after migrating them.

**Reply by j0siepy** [reply]
Thanks for the feedback - we've fixed this 💟

**Comment by thibaudcolas** [comment]
Pretty cool to have this available :)

## Bugs I ran into

- In Windows high contrast mode, the progress bar is invisible.
- The migration progress indicator always seems to stall around 40% of the progress bar.
- Upon reloading the page, I get a "Migration complete" success message with "The old project is still open.View it here". The View it here link isn’t working. Looking at the markup, it’s missing its `href`: `<a target="_blank" rel="noreferrer" class="Link-sc-hrxz1n-0 drSUMK">View it here […]</a>`.
- Even once the migration is complete, Classic projects still display a " This project is currently being migrated. Any updates you make right now will not be migrated.".
- The sort order of cards seems to be lost for PRs. For example, a Classic project’s column has a PR as its second item. After migration, it’s now become the last item in the corresponding Beta project’s column.

Here is a sample project where all of the above happened: [source classic project](https://github.com/wagtail/wagtail/projects/5), [migrated beta project](https://github.com/orgs/wagtail/projects/9/views/1).

## Unexpected behavior

- It creates Private projects, even though the previous ones were public. I’d expect it to create Public projects.
- It added people who had access to the repository as project collaborators, with the same level of access. I think this ones makes complete sense, but it should be documented / have a link once the migration is successful to check who has access. Projects had no access control before, 

## Nice to have

- It would be nice if there was an option to close the previous project, and un-assign everything currently assigned to the project. In the meantime, I did this manually by deleting columns.
- It would be nice if upon successful migration there was a "this project has been migrated, go to the new project" message at the top of the classic project’s board. In the meantime, I also did this manually by adding a new card with a link.

**Reply by mloning-shell** [reply]
> * Even once the migration is complete, Classic projects still display a " This project is currently being migrated. Any updates you make right now will not be migrated.".

Same here and the new project no longer shows the linked PRs on issue cards.

**Comment by jdetaeye** [comment]
Migration of my project seems to hang at the permission step.
![image](https://user-images.githubusercontent.com/5219207/181790913-8d525022-0400-4b0e-9479-832421dacc89.png)

**Reply by jdetaeye** [reply]
Quick update: it eventually finished.   Some form of notification that it may take a while would be useful.

**Reply by j0siepy** [reply]
We've worked on increasing the speed of the migration - but it still might take a little longer if you are migrating a very large project

**Comment by zpc7** [comment]
I can't find the entry 😢 

![image](https://user-images.githubusercontent.com/15316908/182031068-23c539db-e735-48f1-bf09-b7fb17089d24.png)

**Reply by haykam821** [reply]
Did you enable the feature preview for Project Migration?

**Reply by BenjamenMeyer** [reply]
where does one set that `feature preview`? I can't find it in the org settings. Is there a requirement for it (e.g GHE)?

**Reply by j0siepy** [reply]
GHE is not a requirement 💟 
Here is the documentation to enable a feature preview: https://docs.github.com/en/get-started/using-github/exploring-early-access-releases-with-feature-preview

**Reply by BenjamenMeyer** [reply]
Okay - figured it out. I had to enable it on my profile for my associated orgs to be able to use it.

**Comment by Miami87** [comment]
> You can now migrate your old project (classic) boards to the all new projects (beta) experience! Enable "Project migration" in feature preview to try out tables, multiple views, new automation options, and powerful field types using your existing project (classic) boards.
> 
> How it works:
> 
> * We'll create a copy of your existing project (classic) board in Projects (beta)
> * You'll be prompted to close your our old project (classic) as updates made in the new project (Beta) will not be reflected in the closed project (classic)
> * You can reopen the classic project in a pinch if necessary!
> 
> Leave your feedback below!

**Comment by krisu5** [comment]
Bring back markdown cards! [My reasons](https://github.com/orgs/community/discussions/4422#discussioncomment-3415424):

> Without this feature, I will not begin using the new projects.
> It was the number one thing I used in classic projects (even with some flaws) and "drafts" won't cut it.
>In my view, projects and issues are seperate things. And in my repos, using projects for listing issues feels reduntant.
> I rather move to Notion or Trello then start using this. It is a shame, because then less likely people will know that's in the roadmap.

At least make it possible to see the content body and interact with it without opening dialogs.

**Reply by j0siepy** [reply]
We're looking into this! Thank you for the feedback 💟

**Comment by bew** [comment]
### Some information was lost after migration to the new project system.
Previously I had a note referencing an existing issue, with some information ONLY FOR ME:
![image](https://user-images.githubusercontent.com/9730330/186539923-5cc0357b-ba9c-401c-a9d4-9e2943ca6071.png)

With the project conversion, I lost the information that was ONLY FOR ME, and the card is now just the referenced issue:
![image](https://user-images.githubusercontent.com/9730330/186540040-3fee9dbd-a6bf-4376-ab90-1ec0a8de5f95.png)

Clicking on it I have a preview of that issue, but the few lines I wrote for me are not there anymore, and nowhere to be seen :/

**Reply by j0siepy** [reply]
I'm sorry you ran into this issue 💟 If you are looking to recover this information you can still access the old project.

**Reply by bew** [reply]
Yes I know, I went back to the old project and deleted the migrated one.
Was that working as intended? Or is it a bug?
If it was working as intended, is there some place I can see which cards lost information?
What's the expected way to have similar workflow with notes referencing existing issues, with additional information for me?

**Comment by stephen-hqxu** [comment]
It would be nice if you can migrate all classic repo-level projects and merge them into one big project separated by different views because it doesn't make sense to me to split them into multiple projects if they all come from the same repo.

**Reply by j0siepy** [reply]
Thank you for sharing your feedback - We will not be supporting this, as this is not a common request.

**Comment by joewhitsitt** [comment]
Granted it is a really large project board, but we keep getting the angry unicorn:

>This page is taking too long to load.
>Sorry about that. Please try refreshing and contact us if the problem persists.

Wondering if we need to temporarily move some items off the board (~730 issues/cards across 13 columns) to get the rest to migrate.

**Reply by j0siepy** [reply]
You can migrate a total of 1200 items. I would temporarily remove some items from the board and try again.

**Comment by DominickVale** [comment]
Migrating the project for me just creates a project with a bunch of tasks containing only the tags... (e.g "chore", "feature" etc.)

**Reply by j0siepy** [reply]
I need a bit more information in order to help you

**Comment by janpio** [comment]
We could unfortunately not use it, as I don't think we can configure in which field the old "column" is persisted - it is always in "Status". But the status field is special (shown by default on the issue, can not be renamed etc) in new Projects and often has to be used for something else than our old board column.

**Reply by j0siepy** [reply]
I need a bit more information in order to help you. Would you mind sharing a video or diagram to illustrate what you mean?

**Comment by pchelnikov** [comment]
I get error 500 as soon as I press `Start migration` button. Any ideas why?

<img width="765" alt="Screenshot 2023-01-31 at 14 53 48" src="https://user-images.githubusercontent.com/1557594/215779137-cb2a9fb2-2f08-44bf-957a-55bfeadef04a.png">

**Reply by j0siepy** [reply]
We're looking into this, that you for letting us know!

**Reply by j0siepy** [reply]
Should be fixed now!

**Reply by pchelnikov** [reply]
@j0siepy it works now, thanks!

**Comment by SharakPL** [comment]
This feedback is about Feature preview popup in general. There's a video/gif about project migration in the popup, but it's barely visible and there's no way to use fullscreen on the video or change its size in any way. Clicking on a media like that should enlarge it and allow viewing it in 100% scale. All I can do now is open this media alone in new tab.

![obraz](https://user-images.githubusercontent.com/5007456/221354644-05060d0a-1342-4277-ac04-b5c8edf96c97.png)

**Comment by dbalabka** [comment]
Great, but too late. For some reason, the feature preview appeared only recently, but I already manually migrated the project a few months ago. Anyway, thanks for your efforts.

**Comment by asika32764** [comment]
I get unicorn page with message: `This page is taking too long to load` when migrating a project with about 150-200cards.

![screenshot 2023-06-11 下午6 20 20](https://github.com/community/community/assets/1639206/c6d65b36-b68b-4cbb-be2d-f51e64eff1db)

Is there any solutions?

**Comment by bobcoin98** [comment]
enable `Project migration` in Feature preview but there is no option `migrate` 
My project: user project
expected: migration to organization project

**Reply by NCookies** [reply]
I have the same problem.
There are only [Make a copy] and [Close project] even I enabled feature preview.

**Reply by dudwls901** [reply]
I have the same problem.
I'm trying to migrate an organization's project to a user project, but even if I enable project migration in the feature preview, there's no project (classics) menu on the left sidebar, and there's no migration on the option button.

**Comment by JonathanPalmerGD** [comment]
Hey, I just checked on my project today which had 300-500 different cards or so.
It had my project's future work, current work and past accomplishments.
Deleting that github-project kind of destroys my project and motivation to work on it entirely.

I checked the Archived items. Is there any way to recover those cards and migrate them?

Like this event kind of kills my trust in github as a platform, that my project of 5 years was just tanked and destroyed except for like 30 cards.
